### PR TITLE
feat(frontend): add Buy button [GIX-2854][GIX-2705]

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

As initial step for OnRamper onbarding, we create the `Buy` button and a related modal. The modal will be empty for now and the button visible only through a custom env flag `VITE_ONRAMPER_ENABLED`.

Screenshot 2024-09-16 at 09 52 57
Screenshot 2024-09-16 at 09 53 31
Screenshot 2024-09-16 at 09 53 44
Screenshot 2024-09-16 at 09 53 50
Screenshot 2024-09-16 at 09 54 00
Screenshot 2024-09-16 at 09 54 06

### CODE REVIEW

File: src/frontend/src/env/onramper.env.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// TODO: to be removed when we have finished implementing all the OnRamper features
+export const ONRAMPER_ENABLED = JSON.parse(import.meta.env.VITE_ONRAMPER_ENABLED ?? false) === true;
```
Comment (peterpeterparker) [inline src/frontend/src/env/onramper.env.ts:2]: For the state of the art, can you provide the flag in a separate PR?

File: src/frontend/src/lib/components/buy/BuyModal.svelte [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<script lang="ts">
+	import { Modal } from '@dfinity/gix-components';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { modalStore } from '$lib/stores/modal.store';
+</script>
+
+<Modal on:nnsClose={modalStore.close}>
+	<svelte:fragment slot="title"
+		><p class="text-xl">
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/buy/BuyModal.svelte:9]: Shouldn't `<p class="text-xl">` be removed here?

**Comment by AntonioVentilii-DFINITY** [comment]
> Unsolicited feedback: I'm not that convinced by the display on mobile devices

For what I understood we will simplify that and maybe remove the last one, the context menu, to move it somewhere else

File: src/frontend/src/lib/components/buy/BuyModal.svelte (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/buy/BuyModal.svelte]: Good catch! Forgot to remove it!

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Pull request merged**